src/clr-angular/forms/password/password-container.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Component, Input, ContentChild, OnDestroy, Optional, forwardRef, Renderer2 } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { NgControl } from '@angular/forms';
+
+import { IfErrorService } from '../common/if-error/if-error.service';
+import { NgControlService } from '../common/providers/ng-control.service';
+import { LayoutService } from '../common/providers/layout.service';
+import { DynamicWrapper } from '../../utils/host-wrapping/dynamic-wrapper';
+import { ControlIdService } from '../common/providers/control-id.service';
+import { ClrLabel } from '../common/label';
+import { ClarityIcons } from '@clr/icons';
+import { ClrPassword } from './password';
+import { ClrShapeEye, ClrShapeEyeHide } from '@clr/icons/shapes/essential-shapes';
+import { FocusService } from '../common/providers/focus.service';
+import { ControlClassService } from '../common/providers/control-class.service';
+
+@Component({
+  selector: 'clr-password-container',
+  template: `
+    <ng-content select="label"></ng-content>
+    <label *ngIf="!label && addGrid()"></label>
+    <div class="clr-control-container" [ngClass]="controlClass()">
+      <div class="clr-input-wrapper">
+        <div class="clr-input-group" [class.clr-focus]="focus">
+          <ng-content select="[clrPassword]"></ng-content>
+          <clr-icon shape="eye" *ngIf="!show && clrToggle" class="clr-input-group-icon-action" (click)="toggle()"></clr-icon>
+          <clr-icon shape="eye-hide" *ngIf="show && clrToggle" class="clr-input-group-icon-action" (click)="toggle()"></clr-icon>
+        </div>
+        <clr-icon *ngIf="invalid" class="clr-validate-icon" shape="exclamation-circle"></clr-icon>
+      </div>
+      <ng-content select="clr-control-helper" *ngIf="!invalid"></ng-content>
+      <ng-content select="clr-control-error" *ngIf="invalid"></ng-content>
+    </div>
+    `,
+  host: {
+    '[class.clr-form-control]': 'true',
+    '[class.clr-row]': 'addGrid()',
+  },
+  providers: [IfErrorService, NgControlService, ControlIdService, ControlClassService, FocusService],
+})
+export class ClrPasswordContainer implements DynamicWrapper, OnDestroy {
+  subscriptions: Subscription[] = [];
+  invalid = false;
+  control: NgControl;
+  _dynamic = false;
+  show = false;
+  focus = false;
+  @Input() clrToggle = true;
```

Comment:
Nitpick: you should force `show = false` when receiving `false` as an input for `clrToggle`.
It doesn't cost us much, and it will handle weird cases where people allow the user to decide whether the toggle should be displayed or not and the password field would be locked in "clear" mode. I know, it's highly improbable but it's just a couple lines for us.

Reply:
Done.